### PR TITLE
Actually query event logs when verifying event log messages.

### PR DESCRIPTION
**Pull request by jkotalik**
And addresses https://github.com/aspnet/IISIntegration/issues/1062 and https://github.com/aspnet/IISIntegration/issues/1270

Summary of changes:
- Makes file descriptions consistent between inproc, outofproc, shim, and V1.
- Adds process and version information to each event log.
- Adds check to query c# EventLog in tests, rather than relying on the debug log.

### CODE REVIEW

File: src/AspNetCoreModuleV2/CommonLib/EventLog.cpp [hunk 1]
```diff
@@ -13,16 +15,26 @@ EventLog::LogEvent(
     _In_ LPCWSTR pstrMsg
 )
 {
+    static const auto processIdString = GetProcessIdString();
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/EventLog.cpp]: static local? How does it work?

File: src/AspNetCoreModuleV2/CommonLib/debugutil.cpp [hunk 2]
```diff
@@ -21,43 +22,57 @@ inline SRWLOCK g_logFileLock;
...
+    return format(L"File Version: %d.%d.%d.%d. Description: %s",
+        (verInfo->dwFileVersionMS >> 16) & 0xffff,
+        (verInfo->dwFileVersionMS >> 0) & 0xffff,
+        (verInfo->dwFileVersionLS >> 16) & 0xffff,
+        (verInfo->dwFileVersionLS >> 0) & 0xffff,
+        pvProductName);
+}
+
+std::wstring
+GetFileName()
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: GetModuleName

File: src/AspNetCoreModuleV2/CommonLib/debugutil.cpp [hunk 3]
```diff
@@ -21,43 +22,57 @@ inline SRWLOCK g_logFileLock;
...
+    LPVOID lpBuffer = NULL;
+    auto path = GetFileName();
+    DWORD verSize = GetFileVersionInfoSize(path.c_str(), &verHandle);
 
     // Allocate memory to hold data structure returned by GetFileVersionInfo
     std::vector<BYTE> verData(verSize);
 
-    RETURN_LAST_ERROR_IF(!GetFileVersionInfo(path, verHandle, verSize, verData.data()));
-    RETURN_LAST_ERROR_IF(!VerQueryValue(verData.data(), L"\\", &lpBuffer, &size));
+    GetFileVersionInfo(path.c_str(), verHandle, verSize, verData.data());
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: Why did you remove error code checking and signature checking?

File: test/Common.FunctionalTests/Utilities/EventLogHelpers.cs [hunk 4]
```diff
@@ -36,6 +43,58 @@ public static void VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITe
...
+
+            // Perf: only get the last 20 events from the event log.
+            // Eventlog is already sorted based on time of event in ascending time.
+            // Add results in reverse order.
+            for (var i = eventLog.Entries.Count - 1; i >= eventLog.Entries.Count - 20; i--)
+            {
+                eventLogEntries.Add(eventLog.Entries[i]);
+            }
+
+            var expectedRegexEventLog = new Regex(expectedRegexMatchString);
```
Comment (pakrym) [inline test/Common.FunctionalTests/Utilities/EventLogHelpers.cs]: Why create another one?

File: test/Common.FunctionalTests/Utilities/EventLogHelpers.cs [hunk 5]
```diff
@@ -36,6 +43,58 @@ public static void VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITe
...
+
+            var eventLog = new EventLog("Application");
+            var eventLogEntries = new List<EventLogEntry>();
+
+            // Perf: only get the last 20 events from the event log.
+            // Eventlog is already sorted based on time of event in ascending time.
+            // Add results in reverse order.
+            for (var i = eventLog.Entries.Count - 1; i >= eventLog.Entries.Count - 20; i--)
+            {
+                eventLogEntries.Add(eventLog.Entries[i]);
```
Comment (pakrym) [inline test/Common.FunctionalTests/Utilities/EventLogHelpers.cs]: Why copy, just try to match them while enumerating.

File: test/Common.FunctionalTests/Utilities/EventLogHelpers.cs [hunk 6]
```diff
@@ -36,6 +43,58 @@ public static void VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITe
...
+                    eventLogEntry.ReplacementStrings.Length < 3)
+                {
+                    continue;
+                }
+
+                // ReplacementStings == EventData collection in EventLog
+                // This is unaffected if event providers are not registered correctly
+                if (expectedRegexEventLog.IsMatch(eventLogEntry.ReplacementStrings[0]))
+                {
+                    if (processIdString.Contains(eventLogEntry.ReplacementStrings[1]))
```
Comment (pakrym) [inline test/Common.FunctionalTests/Utilities/EventLogHelpers.cs]: Contains? Why not equals?

File: test/Common.FunctionalTests/Utilities/EventLogHelpers.cs [hunk 7]
```diff
@@ -36,6 +43,58 @@ public static void VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITe
...
+            var expectedANCMVersion = "";
+            if (deploymentResult.DeploymentParameters.AncmVersion == AncmVersion.AspNetCoreModule)
+            {
+                expectedANCMVersion = "IIS ASP.NET Core Module";
+            }
+            else if (deploymentResult.DeploymentParameters.AncmVersion == AncmVersion.AspNetCoreModuleV2)
+            {
+                expectedANCMVersion = "IIS ASP.NET Core Module V2";
+            }
+            return description.Contains(expectedANCMVersion);
```
Comment (pakrym) [inline test/Common.FunctionalTests/Utilities/EventLogHelpers.cs]: Why not exact match?

File: test/Common.FunctionalTests/Utilities/EventLogHelpers.cs [hunk 8]
```diff
@@ -36,6 +43,58 @@ public static void VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITe
...
+                    continue;
+                }
+
+                // ReplacementStings == EventData collection in EventLog
+                // This is unaffected if event providers are not registered correctly
+                if (expectedRegexEventLog.IsMatch(eventLogEntry.ReplacementStrings[0]))
+                {
+                    if (processIdString.Contains(eventLogEntry.ReplacementStrings[1]))
+                    {
+                        if (ANCMVersionsMatch(eventLogEntry.ReplacementStrings[2], deploymentResult))
```
Comment (pakrym) [inline test/Common.FunctionalTests/Utilities/EventLogHelpers.cs]: This should be the first condition. No need to try and match events not from ANCM.

File: build/dependencies.props [hunk 9]
```diff
@@ -1,4 +1,4 @@
-﻿<Project>
+<Project>
```
Comment (mikeharder) [inline build/dependencies.props]: Revert?

Comment (jkotalik) [reply build/dependencies.props]: These diffs happen all the time on props/csproj files. I don't bother reverting them most of the time. If you care enough, I'll revert it.

Comment (mikeharder) [reply build/dependencies.props]: Rule of thumb is PRs should not contain any unintended changes.  Recommend instead figuring out why these diffs are happening (maybe file encoding?) and how to fix the issue long term.

Comment (jkotalik) [reply build/dependencies.props]: This is getting inserted: https://en.wikipedia.org/wiki/Byte_order_mark
